src/components/CourseInformation/CourseInformation.js

```diff
@@ -0,0 +1,124 @@
+import { useSelector } from "react-redux";
+
+import CourseImage from "./CourseImage/CourseImage";
+import CourseButton from "./CourseButton/CourseButton";
+import CourseDetails from "./CourseDetails/CourseDetails";
+import CourseDescription from "./CourseDescription/CourseDescription";
+import RelatedCourses from "./RelatedCourses/RelatedCourses";
+
+const CourseInformation = (props) => {
+  // Getting the current location and the data
+  const location = useLocation();
+  const courseData = location.state.expObj;
+  const imgLink = location.state.imgLink;
+
+  // The base url for the back end
+  const api_url = process.env.REACT_APP_ES_MLT_API;
+
+  // Inits the related courses
+  const [relatedCourses, setRelatedCourses] = useState({
+    data: null,
+    isLoading: false,
+    error: null,
+  });
+
+  //  Fetch similar courses from the backend
+  useEffect(() => {
+    setRelatedCourses({ data: null, isLoading: true, error: null });
+
+    // Making call to back end for related courses
+    axios
+      .get(api_url + courseData.meta.id)
+      .then((resp) =>
+        setRelatedCourses({
+          data: resp.data,
+          isLoading: false,
+          error: null,
+        })
+      )
+      .catch((err) => {
+        setRelatedCourses({
+          data: null,
+          isLoading: false,
+          error: err,
+        });
+      });
+  }, [courseData.meta.id, api_url]);
+
+  // Get the global config
+  const { configuration } = useSelector((state) => state.configuration);
+
+  // List of icons that come from the backend
+  const icons = {
+    user: "person-outline",
+    clock: "time-outline",
+    calendar: "calendar-outline",
+    location: "location-outline",
+    hourglass: "hourglass-outline",
+    multi_users: "people-outline",
+  };
+
+  // Returns the icon name to use from backend config
+  const getIconNameToUse = (name) =>
+    icons[name] ? icons[name] : icons["calendar"];
+
+  // Return the value of specific detail.
+  const getCourseDetailValue = (strKey, data) => {
+
+    // gets the keys for the data mapping
+    const objKeys = strKey.split(".");
+
+    // inits with all the data
+    let valueToReturn = data;
+
+
+    // Reduces it down to the specific value
+    objKeys.forEach((key) => {
+      if (valueToReturn) {
+        valueToReturn = valueToReturn[key];
+      }
+    });
+
+    // Returning the desired value.
+    return valueToReturn;
+  };
+
+  // Get the icon to render
+  let courseDetails = null;
+  if (configuration != null){
+    courseDetails = configuration.course_highlights.map((item, index) => {
+      return {
+        icon: getIconNameToUse(item.highlight_icon),
+        name: item.display_name,
+        value: getCourseDetailValue(item.field_name, courseData) || "",
+      };
+    });
+  }
+  return (
+    <div className="content-section">
+      <div className="row content-panel course-detail">
+        <div className="inner-content">
+          <h4>{courseData.Course.CourseTitle}</h4>
```

Comment:
Just to clarify, we're not using the configured course detail fields yet?

Reply:
That is in my branch. I did not want to impede the ability of testing